Actions/Deploy/Deploy.ps1

```diff
@@ -117,66 +140,94 @@ try {
+        }
 
-    try {
-        if ($response.environmentType -eq 1) {
-            if ($bcAuthContext.ClientSecret) {
-                Write-Host "Using S2S, publishing apps using automation API"
-                Publish-PerTenantExtensionApps -bcAuthContext $bcAuthContext -environment $envName -appFiles $apps
-            }
-            else {
+        try {
+            $sandboxEnvironment = ($response.environmentType -eq 1)
+            if ($sandboxEnvironment -and !($bcAuthContext.ClientSecret)) {
+                # Sandbox and not S2S -> use dev endpoint (Publish-BcContainerApp)
+                $parameters = @{
+                    "bcAuthContext" = $bcAuthContext
+                    "environment" = $deploymentSettings.EnvironmentName
+                    "appFile" = $apps
+                }
+                if ($deploymentSettings.SyncMode) {
+                    if (@('Add','ForceSync', 'Clean', 'Development') -notcontains $deploymentSettings.SyncMode) {
+                        throw "Invalid SyncMode $($deploymentSettings.SyncMode) when deploying using the development endpoint. Valid values are Add, ForceSync, Development and Clean."
+                    }
+                    Write-Host "Using $($deploymentSettings.SyncMode)"
+                    $parameters += @{ "SyncMode" = $deploymentSettings.SyncMode }
+                }
                 Write-Host "Publishing apps using development endpoint"
-                Publish-BcContainerApp -bcAuthContext $bcAuthContext -environment $envName -appFile $apps -useDevEndpoint -checkAlreadyInstalled -excludeRuntimePackages
+                Publish-BcContainerApp @parameters -useDevEndpoint -checkAlreadyInstalled -excludeRuntimePackages
             }
-        }
-        else {
-            if ($type -eq 'CD') {
-                Write-Host "Ignoring environment $environmentName, which is a production environment"
+            elseif (!$sandboxEnvironment -and $type -eq 'CD' -and !($deploymentSettings.ContinuousDeployment)) {
+                # Continuous deployment is undefined in settings - we will not deploy to production environments
+                Write-Host "::Warning::Ignoring environment $($deploymentSettings.EnvironmentName), which is a production environment"
             }
             else {
-                # Check for AppSource App - cannot be deployed
+                # Use automation API for production environments (Publish-PerTenantExtensionApps)
+                $parameters = @{
+                    "bcAuthContext" = $bcAuthContext
+                    "environment" = $deploymentSettings.EnvironmentName
+                    "appFiles" = $apps
+                }
+                if ($deploymentSettings.SyncMode) {
+                    if (@('Add','ForceSync') -notcontains $deploymentSettings.SyncMode) {
+                        throw "Invalid SyncMode $($deploymentSettings.SyncMode) when deploying using the automation API. Valid values are Add and ForceSync."
```

Comment:
`Publish-PerTenantExtensionApps` already has `ValidateSet` restriction for `$schemaSyncMode`. Is this really needed?

Reply:
This error message is helpful to the user of AL-Go for GitHub.
If the error comes from further down - people would create a ticket in AL-Go/Issues